Training_25/Program.cs

```diff
@@ -3,11 +3,45 @@
 // Copyright (c) Metamation India.
 // ------------------------------------------------------------------
 // Program.cs
-// Program on main branch.
+// Program prints the Nth Armstrong number.
 // ------------------------------------------------------------------------------------------------
+using static System.Console;
 namespace Training_25;
+
 internal class Program {
    static void Main (string[] args) {
-      Console.WriteLine ("Hello, World!");
+      int n = GetValidN (args);
+      WriteLine (FindNthArmstrong (n));
+   }
+
+   // Gets a valid N from args or prompt
+   static int GetValidN (string[] args) {
+      if (args.Length > 0 && int.TryParse (args[0], out int n) && n is >= 1 and <= 25) return n;
+      while (true) {
+         Write ("Enter N (1 to 25): ");
+         if (int.TryParse (ReadLine (), out n) && n is >= 1 and <= 25) return n;
+         WriteLine ("Invalid input. Try again.");
+      }
+   }
+
+   // Iterates through integers, counts Armstrong numbers, and returns the Nth one.
+   static int FindNthArmstrong (int n) {
+      int count = 0, num = 0;
+      while (true) {
+         if (IsArmstrong (num) && ++count == n) return num;
+         num++;
+      }
+   }
+
+   // Checks if the given number if Armstrong number (sum of its digits each raised to the power
+   // of the total number of digits equals the number itself.)
+   static bool IsArmstrong (int num) {
+      int sum = 0, n = num;

```

Comment:
```suggestion
      int sum = 0, n = num, digit = num.ToString ().Length;

```
Can be initialized in the same line